src/clj/fluree/db/vector/scoring.cljc

```diff
@@ -0,0 +1,37 @@
+(ns fluree.db.vector.scoring
+  #?(:clj (:require [mikera.vectorz.core :as v])))
+
+#?(:clj (set! *warn-on-reflection* true))
+
+(defn vectorize
+  "Takes vector collection v and wraps it in a more optimized vector format
+  for faster vector math."
+  [v]
+  #?(:clj  (v/vec v)
+     :cljs v))
+
+(defn dotproduct
+  [v1 v2]
+  (when (and (vector? v1) (vector? v2))
+    #?(:clj  (v/dot (vectorize v1) (vectorize v2))
```

Comment:
I knew this wasn't super efficient but figured deferring speed optimization could be left for a later effort if needed.

Went ahead and did this now. It required 3 different changes:
a) coercing input values (in-memory flakes hold optimized vectorized objects instead of plain vectors)
b) coercing query `values` - this was probably a bug as I think it should be have been done all along, but this effort exposed it
c) coercing values returned by 'select'